src/platform/ESP32/OTARequestorDriverImpl.h

```diff
@@ -14,23 +14,17 @@
  *    See the License for the specific language governing permissions and
  *    limitations under the License.
  */
-#pragma once
-
-class OTARequesterImpl
-{
-public:
-    static OTARequesterImpl & GetInstance(void)
-    {
-        static OTARequesterImpl instance;
-        return instance;
-    }
 
-    void SendQueryImageCommand(const char * ipAddress, uint32_t nodeId);
+#pragma once
 
-    void SendApplyUpdateRequestCommand(const char * ipAddress, uint32_t nodeId);
+#include <platform/OTARequestorDriver.h>
 
-    void SendNotifyUpdateAppliedCommand(const char * ipAddress, uint32_t nodeId);
+namespace chip {
 
-private:
-    OTARequesterImpl() {}
+class OTARequestorDriverImpl : public OTARequestorDriver
+{
+    bool CheckImageDownloadAllowed();
```

Comment:
Add `override` for methods in `OTARequestorDriver` and use same acess level.

Reply:
Done